src/api/subscription/queries.ts

```diff
@@ -0,0 +1,13 @@
+import { queryClient } from "@/constants/query-client";
+import { setSubscribe } from "./module";
+import { SubscriptionFormData } from "./types";
+
+enum QueryKeys {
+  SUBSCRIBE = "subscribe",
+}
+
+export const fetchSubscribe = async (data: SubscriptionFormData) =>
```

Comment:
for POST methods you don't need useQuery. Please read about mutations https://tanstack.com/query/v4/docs/framework/react/guides/mutations

Reply:
Used mutation